packages/atomic/cypress/integration/result-list/result-components-v1/result-text.cypress.ts

```diff
@@ -0,0 +1,187 @@
+        .shadow()
+        .should('have.text', defaultText);
+    });
+  });
+
+  describe('when the field does not exist for the result and the "default" prop is not set', () => {
+    beforeEach(() => {
+      setupResultTextPage({
+        field: 'thisfielddoesnotexist',
+      });
+    });
+
+    it('should remove the component from the DOM', () => {
+      getFirstResultText().should('not.exist');
+    });
+  });
+
+  describe('when the field value is not a string', () => {
+    beforeEach(() => {
+      setupResultTextPage(
+        {
+          field: 420,
+        },
+        false
+      );
+      interceptSearchResponse((response) =>
+        response.results.forEach((result) => (result.raw['420'] = 'Abc'))
+      );
+      executeFirstSearch();
+    });
+
+    it.skip('should remove the component from the DOM', () => {
+      getFirstResultText().should('not.exist');
+    });
+  });
+
+  describe('when the field value exists & is a string', () => {
+    const field = 'hello_world';
+    const rawValue = 'Hello World!';
+    const localizedValue = 'Welcome!';
+    const highlight: HighlightKeyword = {offset: 6, length: 5};
+    const highlightedValue = rawValue.slice(
+      highlight.offset,
+      highlight.offset + highlight.length
+    );
+    function setupExistingFieldValue(
+      shouldHighlight: boolean,
+      highlightsAvailable: boolean
+    ) {
+      setupResultTextPage(
+        {
+          field: field,
+          'should-highlight': shouldHighlight.toString(),
+        },
+        false
+      );
+      interceptSearchResponse((response) =>
+        response.results.forEach((result) => {
+          result.raw[field] = rawValue;
+          if (highlightsAvailable) {
+            (result as any)[`${field}Highlights`] = [highlight];
+          }
+        })
+      );
+      setFieldCaptions(field, {[rawValue]: localizedValue});
+      executeFirstSearch();
+    }
+
+    describe('when the "shouldHighlight" prop is true and when highlights are available for the field', () => {
+      beforeEach(() => {
+        setupExistingFieldValue(true, true);
+      });
+
+      it('should render the highlighted text', () => {
+        getFirstResultText().should('have.text', rawValue);
+        getFirstResultText()
+          .find('[part="highlight"]')
+          .should('have.text', highlightedValue);
+      });
+    });
+
+    describe('when the "shouldHighlight" prop is false', () => {
+      beforeEach(() => {
+        setupExistingFieldValue(false, true);
+      });
+
+      it('should render text value with localization', () => {
+        getFirstResultText().should('have.text', localizedValue);
+        getFirstResultText().find('[part="highlight"]').should('not.exist');
+      });
+
+      it.skip('should render an "atomic-text" component with the localized value', () => {
+        getFirstResultText()
+          .find('atomic-text')
+          .shadow()
+          .should('have.text', localizedValue);
+      });
```

Comment:
This is another issue I will need to fix. [JIRA](https://coveord.atlassian.net/browse/KIT-993)